opta/core/cloud_client.py

```diff
@@ -28,3 +28,21 @@ def delete_remote_state(self) -> None:
     @abstractmethod
     def get_terraform_lock_id(self) -> str:
         raise NotImplementedError()
+
+    @abstractmethod
+    def list_child_config_names(self) -> List[str]:
+        pass
+
+    @abstractmethod
+    def get_configuration_details(self, config_name: str) -> Dict[str, Any]:
+        pass
+
+    @classmethod
+    def get_config_map(cls) -> Dict[str, List[str]]:
+        pass
+
+    @classmethod
+    def get_detailed_config_map(
+        cls, environment: Optional[str] = None
+    ) -> Dict[str, Dict[str, str]]:
+        pass
```

Comment:
I might be missing something but we should only need **one method to implement per cloud** (to get the content of the files in the remote bucket opta_config).
Every other operation, such as formatting, data manipulation is not cloud specific and could use the same implementation for each cloud.

```
opta config view --aws
```
will do:
1 - call aws.get_config()
2 - (cloud specific) returns content of `opta_config/awsenv-ci` 
3 - render the content to look as opta config files
4 - possible future improvement: do some filtering

only 2 would need to be implemented per cloud.